src/gui/EntryPreviewWidget.cpp

```diff
@@ -31,6 +33,32 @@
 namespace
 {
     constexpr int GeneralTabIndex = 0;
+
+    /*
+     * Display the password in its widget (in color if needed).
+     */
+    void setPassword(QTextEdit* widget, QString password)
+    {
+        if (config()->get(Config::GUI_ColorPasswords).toBool()) {
+            // Show the password in color
+            QString html;
+            for(const auto c : password) {
+                const auto dark = kpxcApp->isDarkTheme();
+                const auto color =
+                    c.isDigit() ? (dark ? "lightblue" : "blue") :
+                    c.isUpper() ? (dark ? "lightgreen" : "green") :
+                    c.isLower() ? (dark ? "yellow" : "red") :
+                        (dark ? "white" : "black");
+                html += "<span style=\"color: " + QString(color) + ";\">" + QString(c).toHtmlEscaped() + "</span>";
```

Comment:
This could be optimised to not repeat the span tag and colour for consecutive characters of the same type. But would that really buy anything?